source/common/router/router.cc

```diff
@@ -306,7 +310,6 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::HeaderMap& headers, bool e
   ASSERT(headers.Scheme());
   ASSERT(headers.Method());
   ASSERT(headers.Host());
-  ASSERT(headers.Path());
```

Comment:
How about just moving all of these lines and the comment to the top?

Reply:
i tried that at first, IIRC if headers.Scheme empty it will be added in the finalizeRequestHeaders function so if we move that assert up i think other modules will break like: https://github.com/envoyproxy/envoy/blob/master/source/extensions/tracers/zipkin/zipkin_tracer_impl.cc#L166

Reply:
Can you take a look to see which other ones can be safely moved earlier?

Reply:
Yeah I would move everything other than Scheme to the beginning of the function along with the comment, then I would move Scheme assert to after where it gets set, with another comment. Thank you!

Reply:
sgtm, i'll push something later tonight